_posts/2019-03-17-three-years-of-react-native.markdown

```diff
@@ -0,0 +1,174 @@
+
+## Keeping up with the Jobs'
+
+As individual developers, it's much less important for us to keep up to date with the latest Apple developer news.
+WWDCs and iOS releases just kinda happen, and we keep moving independently. We still support iOS 9, and have very
+little incentive to bump it - new APIs from Apple just aren't that interesting anymore.
+
+This is a side-effect to the openness of the JavaScript toolchain, and our ability to contribute and fix a lot of
+our own problems at JavaScript level instead of needing to dive deeper into the native side to work on problems.
+
+Without a focused iOS team, it's not easy to pitch for iOS specific projects. This means that features like moving
+to Apple's new app store review API or using iOS features which don't exist on web gets talked about in meetings,
+but never started. This lack of a focused team makes it really hard to implement fixes to bad App Store reviews
+
+That said, in the last year we did manage to ship a pretty hefty [ARKit feature][arkit] - which a positive example
+of an iOS-specific feature which is both forward-thinking tech and a super great fit for Artsy's product. React
+Native basically played no part in that.
+
+## Community Disconnection
+
+There are maybe four communities at play if you're doing React Native:
+
+- Node (Babel/TypeScript/Storybooks/VSCode)
+- React (Hooks/Relay/Styled Components)
+- React Native (Mostly people creating cross-platform tools)
+- Native (Obj-C/Swift/CocoaPods/Fastlane etc)
+
+It's hard to keep on-top of any one community, and it's very hard to keep on track of four. Realistically, if you
+want to be writing apps at the level of quality we do - you need to though.
+
+Being involved in so many communities, as a part of owning [the Omakase][oma] stack means that it's hard to engage
+in the native community with too much excitement anymore. Most of their problems aren't the same as ours.
+
+## Universal Issues?
+
+Are these the kind of problems most teams would have? It depends, with React Native at Artsy our focus on:
+
+- Offering a consistent way to write code across web and iOS, which didn't water down either platform
+- Ensuring we were able to own our dependencies across the stacks
+
+Meant that we took some technical and cultural debt specifically around the platform aspect of our native
+codebase.There are a few levers we can use to fix some of these issues:
+
+- Let iOS be a bit more webby, by using more JS instead of enforcing stricter platform standards
+- Use more of the React Native communities infra-structure, we generally don't use React Native JS dependencies
+  These dependencies usually are cross-platform on Android and iOS which tends to mean making compromises
+  per-platform. Often we are forcing ourselves into extra work to ensure platform consistency
+- Find a way to ensure clearer engineering and product ownership for iOS at Artsy
+- Have a lower mix of iOS infrastructure engineers to product engineers
```

Comment:
Not sure if this means more or less iOS devs.

```suggestion
- Have a lower mix of iOS infrastructure engineers to product engineers.
```

Reply:
Good point, will re-phrase